pycloudlib/cloud.py

```diff
@@ -72,9 +72,9 @@ def __init__(
             name=self.config.get("key_name", user),
         )
         if timestamp_suffix:
-            self.tag = validate_tag(get_timestamped_tag(tag))
+            self.tag = self._validate_tag(get_timestamped_tag(tag))
```

Comment:
After the last modifications, we need to explicitly assign the tag here:

```diff
diff --git a/pycloudlib/cloud.py b/pycloudlib/cloud.py
index e1b7941..375f314 100644
--- a/pycloudlib/cloud.py
+++ b/pycloudlib/cloud.py
@@ -72,9 +72,9 @@ class BaseCloud(ABC):
             name=self.config.get("key_name", user),
         )
         if timestamp_suffix:
-            self.tag = self._validate_tag(get_timestamped_tag(tag))
-        else:
-            self.tag = self._validate_tag(tag)
+            tag = get_timestamped_tag(tag)
+        self._validate_tag(tag)
+        self.tag = tag
 
     def __enter__(self):
         """Enter context manager for this class."""
```

Reply:
oh wow great catch! thank you.